near-sdk/src/utils/stable_map.rs

```diff
@@ -0,0 +1,39 @@
+use std::cell::RefCell;
+use std::collections::BTreeMap;
+
+pub(crate) struct StableMap<K, V> {
```

Comment:
is this struct just for the purpose of providing us with unwrapped/default values? It's internal only, but I feel like whoever has to debug later might find it odd that `get()` modifies state, so feels like a footgun to me

Reply:
It acts as an append-only map (which allows you to keep references to values). This is needed for the caching layer.

Yeah, potentially, this should be documented for future developers, but `get()` only inserts a default value if not existing. All usages have a default value that acts as if it doesn't exist but can be changed in place having a real value. There could be alternative API options, but they would increase complexity with no benefit. Do you have a suggestion on what could be better, or are you suggesting documenting better?

Reply:
yea, just better or more documentation for this